dist/clouduploader.rb

```diff
@@ -31,57 +97,83 @@ def self.credentials(data)
         abort(err)
       end
     end
+    private_class_method :credentials
+  end
 
-    def self.upload(image, data)
-      STDOUT.write("Start uploading image #{image}.\n")
+  class Azure
+    def self.upload(backend_image_file, image_filename, job_data, result_path)
+      image_name = job_data['image_name'].to_s
+      image_name = calculate_image_name(image_filename) if image_name.empty?
+      uncompressed_file = uncompress(backend_image_file)
+      remote_file_name = File.basename(uncompressed_file)
 
-      credentials = credentials(data)
+      run_command("az login",
```

Comment:
Hm, I would prefer to split each of these calls into a dedicated method with a descriptive name, then it would read way easier and I don't need to figure out what the command actually does.

e.g.

```
login
create_storage
upload_image
create_image
...
```